diff -rupN runtime-orig/commands.py runtime/commands.py
--- runtime-orig/commands.py	2013-07-11 16:55:57.292861300 -0300
+++ runtime/commands.py	2013-07-11 16:56:06.157651900 -0300
@@ -187,6 +187,10 @@ class Commands(object):
 
         if sys.platform.startswith('linux'):
             self.osname = 'linux'
+        elif sys.platform.startswith('freebsd'):
+            self.osname = 'linux'
+        elif sys.platform.startswith('cygwin'):
+            self.osname = 'linux'
         elif sys.platform.startswith('darwin'):
             self.osname = 'osx'
         elif sys.platform.startswith('win'):
Binary files runtime-orig/commands.pyc and runtime/commands.pyc differ
diff -rupN runtime-orig/MinecraftDiscovery.py runtime/MinecraftDiscovery.py
--- runtime-orig/MinecraftDiscovery.py	2013-07-11 16:18:43.780957900 -0300
+++ runtime/MinecraftDiscovery.py	2013-07-11 17:04:31.263166700 -0300
@@ -5,6 +5,10 @@ import zipfile
 def getMinecraftPath():
     if   sys.platform.startswith('linux'):
         return os.path.expanduser("~/.minecraft")
+    elif sys.platform.startswith('freebsd'):
+        return os.path.expanduser("~/.minecraft")
+    elif sys.platform.startswith('cygwin'):
+        return os.path.expanduser("~/.minecraft")
     elif sys.platform.startswith('win'):
         return os.path.join(os.getenv("APPDATA"), ".minecraft")
     elif sys.platform.startswith('darwin'):
@@ -16,6 +20,10 @@ def getMinecraftPath():
 def getNativesKeyword():
     if   sys.platform.startswith('linux'):
         return "linux"
+    elif sys.platform.startswith('freebsd'):
+        return "linux"
+    elif sys.platform.startswith('cygwin'):
+        return "linux"
     elif sys.platform.startswith('win'):
         return "windows"
     elif sys.platform.startswith('darwin'):
Binary files runtime-orig/MinecraftDiscovery.pyc and runtime/MinecraftDiscovery.pyc differ
